File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


#define QUARTER 25;
#define DIME 10;
#define NICKEL 5;

int main(void)
{
    float owed = 0;
    int quarter = 0;
    int dime = 0;
    int nickel = 0;
    int moneyLeft = 0;
    int finalAmount = 0;
    int x = 0;
   
   
   do
   { 
        printf("O hai! How much change is owed?\n");
        owed = GetFloat();
   }
   while(owed <= 0);
   
   x = (int)round(owed*100);
   
   quarter = x / QUARTER;
   moneyLeft =  x % QUARTER;
   
   dime =  moneyLeft / DIME;
   moneyLeft = moneyLeft % DIME;
   
   nickel = moneyLeft / NICKEL;
   moneyLeft =  moneyLeft % NICKEL;
   
   
   finalAmount = quarter + dime + nickel + moneyLeft;
   {
       printf("%d\n", finalAmount);
   }
   
    
    return 0;
    
}

File: pset6/greedy.py
print("O hai! How much change is owed?")
while True:
    owed = float(input(""))
    if owed >= 0:
        break
    else:
        print("Invalid input :( Try again")
        continue

owed *= 100
cents = int(owed)
finalAmount= 0

while True:
    if cents >= 25:
        cents -= 25
        finalAmount += 1
    elif ((cents < 25) and (cents >= 10)):
        cents -= 10
        finalAmount += 1
    elif ((cents < 10) and (cents >= 5)):
        cents -= 5
        finalAmount += 1
    elif ((cents < 5) and (cents >= 1)):
        cents -= 1
        finalAmount += 1
    else:
        break

print("" + str(finalAmount))